modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/DataSetDisplay.js

```diff
@@ -393,13 +393,13 @@ function DataSetDisplay({
 	function executeAsyncItemAction(url, method) {
 		return executeAsyncAction(url, method)
 			.then((_) => {
-				delay(500).then(() => {
+				return delay(500).then(() => {
```

Comment:
I [commented on this area previously](https://github.com/liferay-frontend/liferay-portal/pull/376#discussion_r484914922), but in a context like this where you want to return a promise, it would be good to get rid of the explicit `isMounted()` checks, because they are error prone.

For example, `delay()` from [the `useTimeout()` hook](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-react-web/src/main/resources/META-INF/resources/js/hooks/useTimeout.es.js) in `liferay-frontend-js-web` has the check built-in to it. Now, that is calling `setTimeout()` internally and not doing anything with promises, but your `delay()` returns a promise. I don't know if it is only used in functional React components, but if it is, you could make it into a hook.

```js
function useDelay() {
	const isMounted = useIsMounted();

	return useCallback(function delay(ms) {
		return new Promise((resolve) => {
			setTimeout(() => {
				if (isMounted()) {
					resolve();
				}
			}, ms);
		});
	});
}

const delay = useDelay();

doSomething()
	.then(() => delay(500))
	.then(() => doSomethingElse());
```

(Disclaimer: untested, written in a `<textarea>`)